python-project/mortar/image/detector.py

```diff
@@ -0,0 +1,66 @@
+import os
+import cv2
+from typing import Callable
+
+
+class Detector:
+
+    def detect_rects(self,
+                     img_path: str,
+                     fn: Callable[[int, int, int, int], bool],
+                     draw: bool = False) -> list[tuple[int, int, int, int]]:
+        """
+        Detects all rectangles in the provided image.
+
+        Arg img_path is where the image will be loaded from.
+
+        Arg fn will be called like fn(x,y,w,h) with detected rectangles,
+        and is expected to return a bool.
+        If it returns True, that rect will be included in the output.
+
+        Arg draw is optional and defaults to False,
+        It is used mostly for debugging. Will modify self.img with drawn rects.
+        """
+        self.img = cv2.imread(img_path)
+        gray = cv2.cvtColor(self.img, cv2.COLOR_BGR2GRAY)
+        _, thresh = cv2.threshold(gray, 50, 255, 0)
+        contours, _ = cv2.findContours(thresh, 1, 2)
```

Comment:
The `mode` and `method` arguments are `[RetrievalModes](https://docs.opencv.org/3.4/d3/dc0/group__imgproc__shape.html#ga819779b9857cc2f8601e6526a3a5bc71)` and `[ContourApproximationModes](https://docs.opencv.org/3.4/d3/dc0/group__imgproc__shape.html#ga4303f45752694956374734a03c54d5ff)` respectively. Can you change the call to use those names instead of `1, 2` so the reader doesn't have to go look into the OpenCV source code to understand the meaning of those arguments in the call?